syft/source/directory_resolver.go

```diff
@@ -244,10 +259,21 @@ func (r directoryResolver) addSymlinkToIndex(p string, info os.FileInfo) (string
 		return "", fmt.Errorf("unable to readlink for path=%q: %w", p, err)
 	}
 
-	// note: if the link is not absolute (e.g, /dev/stderr -> fd/2 ) we need to resolve it relative to the directory
-	// in question (e.g. resolve to /dev/fd/2)
-	if !filepath.IsAbs(linkTarget) {
+	if filepath.IsAbs(linkTarget) {
+		// if the link is absolute (e.g, /bin/ls -> /bin/busybox) we need to
+		// resolve relative to the root of the base directory
+		linkTarget = filepath.Join(r.base, linkTarget)
+	} else {
+		// if the link is not absolute (e.g, /dev/stderr -> fd/2 ) we need to
+		// resolve it relative to the directory in question (e.g. resolve to
+		// /dev/fd/2)
 		linkTarget = filepath.Join(filepath.Dir(p), linkTarget)
+
+		// we then need to ensure that the relative link cannot escape from the
+		// base directory
+		if !strings.HasPrefix(linkTarget, r.base) {
```

Comment:
Note to self: this is probably not a great idea - it feels like we could confuse `/foo` and `/foobar` here?